src/transitions/modal-transition.js

```diff
@@ -0,0 +1,249 @@
+/**
+ * @author Deux Huit Huit
+ *
+ * Popup page transition
+ *
+ */
+(function ($, undefined) {
+	
+	'use strict';
+	
+	var win = $(window);
+	var body = $('body');
+	var sitePages = $('#site-pages');
+	var bgTransition = $('#bg-transition');
+	var bgTransitionModal = $('#bg-transition-modal');
+	var DEFAULT_DELAY = 350;
+	var MODAL_SELECTOR = '.js-modal-page';
+
+	var beginCompleted = false;
+	var loadCompleted = false;
+
+	var dataIn = null;
+	
+	/* ENTERING FUNCTIONS */
+	var completeAnim = function (data, callback) {
+		var leavingPage = data.currentPage;
+		var enteringPage = data.nextPage;
+		
+		var domEnteringPage = $(enteringPage.key());
+		var domLeavingPage = $(leavingPage.key());
+		
+		var modal = domEnteringPage.find(MODAL_SELECTOR);
+		
+		// Leave the current page
+		leavingPage.leave(data.leaveCurrent, {
+			canRemove: false
+		});
+		
+		//deactivate toggles in enter modal
+		domEnteringPage.find('a[data-action="toggle"]')
+			.removeAttr('data-action')
+			.attr('href', domLeavingPage.attr('data-response-url'));
+		
+		// Make the page hidden for assistive technology
+		domLeavingPage.attr('aria-hidden', 'true').attr('role', 'presentation');
+		domEnteringPage.removeAttr('aria-hidden').removeAttr('role');
+		
+		// Add body class
+		body.addClass(enteringPage.key().substring(1));
+		
+		// Notify entering page
+		App.modules.notify('page.entering', {page: enteringPage, route: data.route});
+		
+		// Animate leaving and start entering after leaving animation
+		// Need a delay for get all Loaded
+		domEnteringPage.ready(function () {
+			// close modal if already opened
+			App.modules.notify('modal.prepare', {
+				item: modal
+			});
+			domEnteringPage.css({
+				opacity: 1,
+				display: 'block',
+				position: 'relative'
+			}).height();
+			enteringPage.enter(data.enterNext);
+			App.modules.notify('modal.show', {
```

Comment:
Is there a module that listens to modal.show/prepare/etc. ?

Reply:
Need to validate. all the transition in this file seem depend on an other module.

Reply:
Checked and is in auto-modal module.